### PR TITLE
Add AbstractHazelcastInstanceProxy to avoid duplication

### PR DESCRIPTION
**Pull request by pveentjer**
There are a set of HazelcastInstance implementations that are proxies and do not do much more than forwarding to request to some target object. This leads to a lot of code duplication.

The AbstractHazelcastInstanceProxy takes care of the forwarding logic so that the proxies can concentrate on adding additional behavior.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/instance/impl/BootstrappedInstanceProxy.java [hunk 1]
```diff
@@ -16,64 +16,27 @@
...
 import javax.annotation.Nonnull;
-import java.util.Collection;
 import java.util.List;
-import java.util.UUID;
-import java.util.concurrent.ConcurrentMap;
 
 // A special HazelcastInstance that has a BootstrappedJetProxy
 @SuppressWarnings({"checkstyle:methodcount"})
-public final class BootstrappedInstanceProxy implements HazelcastInstance {
+public final class BootstrappedInstanceProxy extends AbstractHazelcastInstanceProxy {
```
Comment (srknzl) [inline hazelcast/src/main/java/com/hazelcast/instance/impl/BootstrappedInstanceProxy.java:30]: nit: for the warning

```suggestion
public final class BootstrappedInstanceProxy extends AbstractHazelcastInstanceProxy<HazelcastInstance> {
```

File: hazelcast/src/main/java/com/hazelcast/instance/impl/AbstractHazelcastInstanceProxy.java [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    public AbstractHazelcastInstanceProxy(H target) {
+        this.target = target;
+    }
+
+    /**
+     * Sets the target instance which could be <code>null</code>.
+     *
+     * @param target the target instance.
+     */
+    public void target(H target) {
```
Comment (srknzl) [inline hazelcast/src/main/java/com/hazelcast/instance/impl/AbstractHazelcastInstanceProxy.java:84]: nit: wont setTarget be more explicit?

File: hazelcast/src/main/java/com/hazelcast/osgi/impl/HazelcastOSGiInstanceImpl.java [hunk 3]
```diff
@@ -16,295 +16,32 @@
...
-import java.util.UUID;
-import java.util.concurrent.ConcurrentMap;
 
 /**
  * {@link com.hazelcast.osgi.HazelcastOSGiInstance} implementation
  * as proxy of delegated {@link com.hazelcast.core.HazelcastInstance} for getting from OSGi service.
  */
 @SuppressWarnings({"checkstyle:classfanoutcomplexity"})
-class HazelcastOSGiInstanceImpl
+class HazelcastOSGiInstanceImpl extends AbstractHazelcastInstanceProxy
```
Comment (srknzl) [inline hazelcast/src/main/java/com/hazelcast/osgi/impl/HazelcastOSGiInstanceImpl.java:31]: nit: 
```suggestion
class HazelcastOSGiInstanceImpl extends AbstractHazelcastInstanceProxy<HazelcastInstance>
```

**Review by srknzl — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/instance/impl/AbstractHazelcastInstanceProxy.java (hunk 2 above)
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/instance/impl/AbstractHazelcastInstanceProxy.java:84]: Done

**Review by TomaszGaweda — APPROVED** [review]

**Pull request closed without merging**